src/main/java/app/services/impl/GameServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package app.services.impl;
+
+import app.dtos.GameDTO;
+import app.services.GameService;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashSet;
+import java.util.Set;
+
+public class GameServiceImpl implements GameService {
+    private final Logger logger = LoggerFactory.getLogger(GameServiceImpl.class);
+
+    private final String BASE_URL = "https://api.rawg.io/api/games";
+    private final String API_KEY = System.getenv("API_KEY");
+
+    private final ObjectMapper objectMapper;
+
+    public GameServiceImpl(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        objectMapper.registerModule(new JavaTimeModule());
+    }
+
+    @Override
+    public Set<GameDTO> getAllGames() {
+        int amountOfGames = 25;
+        String uri = String.format("%s?key=%s&page_size=%s", BASE_URL, API_KEY, amountOfGames);
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .header("Accept", "Application/json")
+                .uri(URI.create(uri))
+                .GET()
+                .build();
+
+        try {
+            HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
Maybe the HttpClient should be reused by storing it in a class variable and using it for each method. I believe the HttpClient should be able to be shared between multiple requests without issue.